### DISCUSSION TITLE
'require' call may be converted to import

### DISCUSSION THREAD

**Discussion by dklymenk** [infinitered/ignite#1786] [Q&A] [answered]
In a freshly ignited project there are code action suggestions in a few files that declare images like this:
```js
export const logoIgnite = require("./logo-ignite.png")
```
It looks like this:
![DeepinScreenshot_select-area_20210902101553](https://user-images.githubusercontent.com/64093836/131800571-3732c56e-6002-459e-b96a-0af6d2a811ee.png)

I'm pretty new to typescript, so not sure if my previous experience applies here, but I would assume that if code action is available, it means that there is something at least sub-optimal in your code.

Funnily enough, when I take the code action, I get diagnostic errors, yet since those are not compiler errors the code still runs just fine.
![DeepinScreenshot_select-area_20210902102708](https://user-images.githubusercontent.com/64093836/131801076-d2443d4e-822d-4817-813f-df6aaeac0364.png)

Is there something one can add to `tsconfig.json` to remedy this? I am using https://github.com/typescript-language-server/typescript-language-server. I did a quick sanity check and I get the same diagnostics in vscode, both vscodium and the proprietary one. 

I read a bunch of issues online where users report annoying code action suggestions like "This may be converted to an async function" that they can't silence or deal with. So if there is no oversight with `tsconfig.json` in this repo and there is nothing you can do with that suggestion, my question would be is this the intended way to do typescript where you have code actions available and never take them and they are clogging up your diagnostics? 

EDIT: Fix references to `tsconfig.json`.

**Comment by jamonholmgren** [comment, marked as answer]
@dklymenk This is a reasonable question, for sure. React Native uses Metro, which is a bundler, and as far as I know, Metro requires (heh) that we use `require` to pull in images. It's interesting to me that `import` worked at all, there.

If you know of a way to globally disable this rule/code action, please let me know, and I'd love a PR!

**Comment by dklymenk** [comment]
Thanks for your response.

Actually there is no way to disable these rules. This issue is pretty close to becoming a meme at this point. Some of these code actions are known to either break the code or do something unreasonable like making a function async for no actual reason.
https://github.com/microsoft/TypeScript/issues/28825
https://github.com/microsoft/TypeScript/issues/37332

If you're curious I ended up forking typescript and adding support for filtering those error codes myself. That entire project is a huge mess, there is no centralized place where you can filter all the errors, so I ended up creating a higher order function that filters the codes.

https://github.com/dklymenk/TypeScript/commit/07c294266b29dc1ecb144a384a1697a1b88a60e8

It's nowhere near a PR status as error codes are hard-coded and I haven't had time to figure out how to fetch them from project's tsconfig, but it works great for me, even though it's kinda annoying to recompile it every time I want to add a new suggestion to the ignore list.

**Reply by jamonholmgren** [reply]
Woowwww. That's intense!
